src/lib/urlHelper.js

```diff
@@ -34,34 +36,52 @@ export function stripProtocol(url) {
  */
 const uriChars = /[\w\-.~]/i;
 const ucsChars = /[\xA0-\u{D7FF}\u{F900}-\u{FDCF}\u{FDF0}-\u{FFEF}\u{10000}-\u{1FFFD}\u{20000}-\u{2FFFD}\u{30000}-\u{3FFFD}\u{40000}-\u{4FFFD}\u{50000}-\u{5FFFD}\u{60000}-\u{6FFFD}\u{70000}-\u{7FFFD}\u{80000}-\u{8FFFD}\u{90000}-\u{9FFFD}\u{A0000}-\u{AFFFD}\u{B0000}-\u{BFFFD}\u{C0000}-\u{CFFFD}\u{D0000}-\u{DFFFD}\u{E1000}-\u{EFFFD}]/u;
-const validIRIChar = (char) => (uriChars.test(char) || ucsChars.test(char));
-// `sanitizeIRI` does not actually URI-encode the chars (that is the browser's and server's job), just removes the ones that are not allowed.
-export function sanitizeIRI(str, { replacement = "" } = {}) {
-  if (!isString(str)) throw "The input slug must be a string.";
-  if (!isString(replacement)) throw "`options.replacement` must be a string.";
+const validURIChar = char => uriChars.test(char);
+const validIRIChar = char => uriChars.test(char) || ucsChars.test(char);
+// `sanitizeURI` does not actually URI-encode the chars (that is the browser's and server's job), just removes the ones that are not allowed.
+export function sanitizeURI(str, { replacement = "", encoding = "unicode" } = {}) {
+  if (!isString(str)) {
+    throw new Error("The input slug must be a string.");
+  }
+  if (!isString(replacement)) {
+    throw new Error("`options.replacement` must be a string.");
+  }
+  
+  let validChar;
+  if (encoding === "unicode") {
+    validChar = validIRIChar;
+  } else if (encoding === "ascii") {
+    validChar = validURIChar;
+  } else {
+    throw new Error('`options.encoding` must be "unicode" or "ascii".');
+  }
 
   // Check and make sure the replacement character is actually a safe char itself.
-  if (!Array.from(replacement).every(validIRIChar)) throw "The replacement character(s) (options.replacement) is itself unsafe.";
+  if (!Array.from(replacement).every(validChar)) {
+    throw new Error("The replacement character(s) (options.replacement) is itself unsafe.");
+  }
 
   // `Array.from` must be used instead of `String.split` because
   //   `split` converts things like emojis into UTF-16 surrogate pairs.
-  return Array.from(str).map(char => (validIRIChar(char) ? char : replacement)).join('');
+  return Array.from(str).map(char => (validChar(char) ? char : replacement)).join('');
 }
 
-export function sanitizeSlug(str, { replacement = '-' } = {}) {
-  if (!isString(str)) throw "The input slug must be a string.";
-  if (!isString(replacement)) throw "`options.replacement` must be a string.";
+export function sanitizeSlug(str, options = Map()) {
+  const encoding = options.get('encoding', 'unicode');
+  const stripDiacritics = options.get('clean_accents', false);
+  const replacement = options.get('sanitize_replacement', '-');
```

Comment:
Intentionally undocumented?

Reply:
I wasn’t sure whether we wanted to wait or not, thoughts?

Reply:
I wasn't sure if we wanted to wait until someone actually had a valid use case for it -- validating it in `src/actions/config.js` would take a bit of effort. Thoughts?

Reply:
Nah, I'm fine with it as is, just was curious.